main.py

```diff
@@ -18,24 +19,25 @@ def signup():
     password = request.form["password"]
     verify = request.form["verify"]
     email = request.form["email"]
-    string_len = True
+    
+    username_error = ""
+    password_error = ""
+    verify_error = ""
+    email_error = ""
+    
     if username == "" or check_character_len(username) == False:
         username_error = "That's not a valid username"
-    else:
-        username_error = ""
+
     if password == "" or check_character_len(password) == False:
         password_error = "That's not a valid password"
-    else:
-        password_error = ""
+
     if verify == "" or verify != password or check_character_len(verify) == False:
         verify_error = "Passwords don't match"
-    else:
-        verify_error = ""
+
     if email != "" and validate_email(email) == False:
```

Comment:
This doesn't meet the "acceptance criteria".

`Note: the email field may be left empty, but if there is content in it, then it must be validated.` from https://coding102.devetry.com/assignments/user-signup/